app/models/concerns/authenticator.rb

```diff
@@ -0,0 +1,50 @@
+module Authenticator
+  
+  extend ActiveSupport::Concern
+  
+  def verify_email
+    update(verified_at: Time.current, verification_token: nil)
+  end
+
+  def activated?
+    verified_at.present?
+  end
+
+  def authenticated?(attribute, token)
+    digest = send("#{attribute}_digest")
+    return false if digest.nil?
+    Tokenizer.is_digest_of?(digest, token)
+  end
+
+  def set_reset_digest
+    self.reset_token = Tokenizer.new_token
+    update(reset_digest: Tokenizer.create_digest(reset_token), reset_sent_at: Time.current)
+  end
+
+  def send_password_reset_email
+    SendPasswordResetEmailJob.perform_later(id, reset_token)
+  end
+
+  def password_reset_expired?
+    reset_sent_at < RESET_TOKEN_EXPIRY_TIME.ago
+  end
+
+  private
+
+    def send_verification_email
+      SendVerificationEmailJob.perform_later(id) if customer?
+    end
+
+    def set_verification_token
+      if verification_token.present?
```

Comment:
can be done in a single line return false if verification_token.present?

Reply:
done